src/components/authentication/RegisterPage.vue

```diff
@@ -3,70 +3,100 @@
 
 		<div class="register-page__spacer"/>
 
-		<v-form ref="form" lazy-validation>
+		<v-form @submit="checkFields()" ref="form" v-model="valid" lazy-validation>
 			<v-text-field
-				v-model="credential.name"
-				label="Name/Email"
+				v-model="user.email"
+				:rules="emailRules"
+				label="Email"
+				type="email"
 				required
 			/>
 			<v-text-field
-				v-model="credential.password"
+				v-model="user.password"
+				:rules="passwordRules"
 				label="Password"
+				type="password"
 				required
 			/>
 			<v-text-field
 				v-model="confirm_password"
+				:rules="passwordRules"
 				label="Confirm Password"
+				type="password"
 				required
 			/>
 			<div class="login-page__form__buttons">
-				<v-btn @click="checkFields()" class="login-page__form__buttons__btn">
-					Register Me
+				<v-btn
+					type="submit"
+					class="login-page__form__buttons__button"
+					:disabled="!valid"
+				>
+					Signup
 				</v-btn>
-				<router-link :to="{name: 'LoginPage'}">
-					<v-btn>already have an account</v-btn>
+				<router-link
+					:to="{name: 'login-page'}"
+					class="login-page__form__buttons__button"
+				>
+					<v-btn>have an account</v-btn>
 				</router-link>
 			</div>
 		</v-form>
-
 	</div>
 </template>
 
 <script>
 	import UserRepository from '../../classes/user/UserRepository.js'
+	import User from '../../classes/user/User'
 
 	export default {
 		name: 'RegisterPagePage',
 		components: {},
 		data() {
 			return {
-				credential: {
-					name: '',
-					password: ''
-				},
-				confirm_password: ''
+				user:  new User,
```

Comment:
user : {
type: User,
}
?

Reply:
ok, i done that only for good view that objects in db.json (with all fields)